protocol/x/clob/e2e/permissioned_keys_test.go

```diff
@@ -662,3 +662,758 @@ func TestPlaceOrder_PermissionedKeys_Failures(t *testing.T) {
+			},
+		},
+		"Short term maker order is removed if permissioned key is removed": {
+			smartAccountEnabled: true,
+			blocks: []TestBlockWithMsgs{
+				{
+					Block: 2,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: []sdk.Msg{
+								&aptypes.MsgAddAuthenticator{
+									Sender:            constants.BobAccAddress.String(),
+									AuthenticatorType: "AllOf",
+									Data:              compositeAuthenticatorConfig,
+								},
+							},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        300_000,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{1},
+							Signers:    []cryptotypes.PrivKey{constants.BobPrivateKey},
+
+							ExpectedRespCode: 0,
+						},
+					},
+				},
+				{
+					Block: 4,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: []sdk.Msg{
+								clobtypes.NewMsgPlaceOrder(
+									testapp.MustScaleOrder(
+										constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20,
+										testapp.DefaultGenesis(),
+									),
+								),
+							},
+							Authenticators: []uint64{0},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        0,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{0},
+							// Sign using Alice's private key.
+							Signers: []cryptotypes.PrivKey{constants.AlicePrivateKey},
+
+							ExpectedRespCode: 0,
+						},
+					},
+				},
+				{
+					Block: 6,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: []sdk.Msg{
+								&aptypes.MsgRemoveAuthenticator{
+									Sender: constants.BobAccAddress.String(),
+									Id:     0,
+								},
+							},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        300_000,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{2},
+							Signers:    []cryptotypes.PrivKey{constants.BobPrivateKey},
+
+							ExpectedRespCode: 0,
+						},
+					},
+				},
+				{
+					Block: 8,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: []sdk.Msg{
+								clobtypes.NewMsgPlaceOrder(
+									testapp.MustScaleOrder(
+										constants.Order_Alice_Num0_Id1_Clob1_Sell5_Price15_GTB20_IOC,
+										testapp.DefaultGenesis(),
+									),
+								),
+							},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        0,
+							AccountNum: []uint64{0},
+							SeqNum:     []uint64{0},
+							Signers:    []cryptotypes.PrivKey{constants.AlicePrivateKey},
+
+							ExpectedRespCode: 0,
+						},
+					},
+				},
+			},
+			expectedOrderIdsInMemclob: map[clobtypes.OrderId]bool{
+				constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20.OrderId:       false,
+				constants.Order_Alice_Num0_Id1_Clob1_Sell5_Price15_GTB20_IOC.OrderId: false,
+			},
+			expectedOrderFillAmounts: map[clobtypes.OrderId]uint64{
+				constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20.OrderId:       0,
+				constants.Order_Alice_Num0_Id1_Clob1_Sell5_Price15_GTB20_IOC.OrderId: 0,
+			},
+		},
+		"Stateful maker order can be matched even if permissioned key is removed": {
```

Comment:
Could we provide some context on these behaviors when the order is removed? My understanding (please CWIW): 
- For ST orders, we HAVE to remove the maker order because we can't trust BP on this maker order being valid
- For stateful orders, we don't HAVE TO remove the maker order because we can trust the onchain state. We MAY choose to remove the maker stateful order, but needs additional work (e.g. prune orders when removing authenticators)

Reply:
yeah that's correct. for ST orders, it's not necessarily about trust, it's more about maintaining the invariant that operations queue is always valid.

Reply:
Thanks, to clarify I was suggesting to comment on the test cases about this difference in expected behavior between ST and stateful order